lib/authentication/auth_web.js

```diff
@@ -98,12 +99,28 @@ function AuthWeb(connectionConfig, httpClient, webbrowser) {
       throw new Error(util.format('Invalid SSO URL found - %s ', loginUrl));
     }
 
-    // Step 3: open browser
-    open(loginUrl);
+    const currentTime = Date.now();
+    while (browserActionTimeout === 0 || Date.now() < currentTime + browserActionTimeout) {
+      try {
+        // Step 3: open browser
+        open(loginUrl);
 
-    // Step 4: get SAML token
-    const tokenData = await withBrowserActionTimeout(browserActionTimeout, receiveData);
-    processGet(tokenData);
+        // Step 4: get SAML token
+        const tokenData = await withBrowserActionTimeout(browserActionTimeout, receiveData);
```

Comment:
I think we need to reduce the browserActionTimeout each time otherwise in the worst case the actual timeout could be doubled.